lambdas/pkgpush/index.py

```diff
@@ -360,3 +418,127 @@ def get_pkg(src_registry, data):
         pkg_max_size=PKG_FROM_FOLDER_MAX_PKG_SIZE,
         pkg_max_files=PKG_FROM_FOLDER_MAX_FILES,
     )
+
+
+def large_request_handler(request_type):
+    user_request_key = f'user-requests/{request_type}'
+
+    def inner(f):
+        @functools.wraps(f)
+        def wrapper(request):
+            version_id = request.data
+            size = s3.head_object(Bucket=SERVICE_BUCKET, Key=user_request_key, VersionId=version_id)['ContentLength']
+            if size > LAMBDA_TMP_SPACE:
+                raise ApiException(
+                    HTTPStatus.BAD_REQUEST,
+                    f'Request file size is {size}, '
+                    f'but max supported size is {LAMBDA_TMP_SPACE}.'
+                )
+            # download file with user request using lambda's role
+            with tempfile.TemporaryFile() as tmp_file:
+                s3.download_fileobj(
+                    SERVICE_BUCKET,
+                    user_request_key,
+                    tmp_file,
+                    ExtraArgs={'VersionId': version_id},
+                )
+                tmp_file.seek(0)
+                request.stream = tmp_file
+                result = f(request)
+                try:
+                    s3.delete_object(SERVICE_BUCKET, user_request_key, version_id)
+                except Exception:
+                    logger.exception('error while removing user request file from S3')
+                return result
+        return wrapper
+    return inner
+
+
+@api(cors_origins=get_default_origins(), request_class=ELBRequest)
+@api_exception_handler
+@auth
+@json_api({'type': 'string', 'minLength': 1, 'maxLength': 1024})
+@large_request_handler('create-package')
+@setup_telemetry
+def create_package(request):
+    json_iterator = map(json.JSONDecoder().decode, (line.decode() for line in request.stream))
+
+    data = next(json_iterator)
```

Comment:
I'd suggest calling this `first_line` or something because the first line is special. `data` makes more sense if you're going to iterate with it.